src/rhsmlib/services/entitlement.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import print_function, division, absolute_import
+
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+import logging
+
+from subscription_manager import injection as inj
+from subscription_manager.i18n import ugettext as _
+from subscription_manager import managerlib, utils
+
+from rhsmlib.services import exceptions, products
+
+log = logging.getLogger(__name__)
+
+
+class EntitlementService(object):
+    def __init__(self, cp=None):
+        self.cp = cp
+        self.identity = inj.require(inj.IDENTITY)
+        self.product_dir = inj.require(inj.PROD_DIR)
+        self.entitlement_dir = inj.require(inj.ENT_DIR)
+
+    def get_status(self, on_date=None):
```

Comment:
I'm not totally sure I understand your question.  If you're asking why we don't have a PluginManager instance in this class, it's because we don't expose status listing as an event that plugins can hook on to.  Our plugin framework is not really used very extensively either, so I don't see a pressing need to add a new conduit that will expose status queries to plugins.